### PR TITLE
ostatnie

### PR DESCRIPTION
**Pull request by OlafKozak**
pięknie

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @OlafKozak

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#include "vectorFunctions.hpp"
+#include <vector>
+#include <iostream>
+
+std::vector<std::shared_ptr<int>> generate(int count){
+    std::vector<std::shared_ptr<int>> v;
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:6]: Reserve memory when you know how many elements will be inside vector.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    for(int i=0;i<count;i++){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:7]: Add space after `if`, `for`, `while` `function` etc...

Add space between operator like `+` `-` `/` `:` `&&` etc..

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        v.push_back(std::make_shared<int>(i));
+    }
+    return v;
+}
+
+void print(std::vector<std::shared_ptr<int>> v){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:13]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    for(auto i : v){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:14]: const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        std::cout << *i << std::endl;
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:15]: Ptr may be null

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    }
+}
+
+void add10(std::vector<std::shared_ptr<int>> v){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:19]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    for(int i=0;i<v.size();i++){
+        if(v[i] != nullptr) *v.at(i).get() += 10;
+    }
+}
+
+void sub10(int *a){
+    if(a != nullptr) *a -= 10;
+}
+
+void sub10(std::vector<std::shared_ptr<int>> v){
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:29]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+void add10(std::vector<std::shared_ptr<int>> v){
+    for(int i=0;i<v.size();i++){
+        if(v[i] != nullptr) *v.at(i).get() += 10;
+    }
+}
+
+void sub10(int *a){
+    if(a != nullptr) *a -= 10;
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:26]: One command one line

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+void print(std::vector<std::shared_ptr<int>> v){
+    for(auto i : v){
+        std::cout << *i << std::endl;
+    }
+}
+
+void add10(std::vector<std::shared_ptr<int>> v){
+    for(int i=0;i<v.size();i++){
+        if(v[i] != nullptr) *v.at(i).get() += 10;
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:21]: One command one line

Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:21]: you don't need get. If you use *vector[i] it is enough

Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:21]: if you use .at you should also handle exception for std:out_of_range

**Review by nauka-programowania-MA — CHANGES_REQUESTED** [review]

**Pull request closed without merging**